pkg/controller/dataimportcron-controller.go

```diff
@@ -1036,6 +1024,21 @@ func addDataImportCronControllerWatches(mgr manager.Manager, c controller.Contro
 		return err
 	}
 
+	mapCronJobToCron := func(_ context.Context, obj client.Object) []reconcile.Request {
+		return []reconcile.Request{{NamespacedName: types.NamespacedName{Namespace: obj.GetLabels()[common.DataImportCronNsLabel], Name: getCronName(obj)}}}
+	}
+
+	if err := c.Watch(source.Kind(mgr.GetCache(), &batchv1.CronJob{}),
+		handler.EnqueueRequestsFromMapFunc(mapCronJobToCron),
+		predicate.Funcs{
+			CreateFunc: func(e event.CreateEvent) bool { return getCronName(e.Object) != "" },
```

Comment:
So only triggering on creation is enough?

Reply:
Yep, to fix this issue we only care specifically about the first time the cronjob is visibile to the new instance of DIC controller. This is "create" although cronjob might have been created long time before.